hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/index/simple/SparkHoodieSimpleIndex.java

```diff
@@ -147,6 +150,12 @@ public boolean isImplicitWithStorage() {
     JavaSparkContext jsc = HoodieSparkEngineContext.getSparkContext(context);
     int fetchParallelism = Math.max(1, Math.max(baseFiles.size(), parallelism));
     return jsc.parallelize(baseFiles, fetchParallelism)
-        .flatMapToPair(partitionPathBaseFile -> new HoodieKeyLocationFetchHandle(config, hoodieTable, partitionPathBaseFile).locations());
+        .flatMapToPair( partitionPathBaseFile -> {
+          Iterator<ImmutablePair<HoodieKey, HoodieRecordLocation>> iter =
```

Comment:
is there a way to do this lazily? convert to a stream and map to tuple and get the underlying iterator?